### PR TITLE
Follow camera improvements

### PR DESCRIPTION
**Pull request by pawelbudziszewski**
Introduced 2 improvements to the `follow camera` component:

- Ability to define min and max camera distance from the UI.
- Added `Look at center entity` UI option. When activated, the camera will be rotated to look at the center entity. It makes it more convenient to set camera rotation this way.

### CODE REVIEW

File: Project/Gem/Source/KrakenCamera/FollowingCameraComponent.cpp [hunk 1]
```diff
@@ -1,20 +1,19 @@
 /*
-* Copyright (c) Contributors to the Open 3D Engine Project.
-* For complete copyright and license terms please see the LICENSE at the root of this distribution.
-*
-* SPDX-License-Identifier: Apache-2.0 OR MIT
-*
-*/
+ * Copyright (c) Contributors to the Open 3D Engine Project.
```
Comment (michalpelka) [inline Project/Gem/Source/KrakenCamera/FollowingCameraComponent.cpp:2]: Unnecessary change, please remove it for lines 1-7

File: Project/Gem/Source/KrakenCamera/FollowingCameraComponent.cpp [hunk 2]
```diff
@@ -29,7 +28,10 @@ namespace AppleKraken
                 ->Field("Target", &FollowingCameraComponent::m_target)
                 ->Field("SmoothingLength", &FollowingCameraComponent::m_smoothingBuffer)
                 ->Field("ZoomSpeed", &FollowingCameraComponent::m_zoomSpeed)
-                ->Field("RotationSpeed", &FollowingCameraComponent::m_rotationSpeed);
+                ->Field("RotationSpeed", &FollowingCameraComponent::m_rotationSpeed)
+                ->Field("MinCameraDistance", &FollowingCameraComponent::m_min_camera_distance)
```
Comment (michalpelka) [inline Project/Gem/Source/KrakenCamera/FollowingCameraComponent.cpp:32]: Convention - please use camel-case instead '_'

File: Project/Gem/Source/KrakenCamera/FollowingCameraComponent.cpp [hunk 3]
```diff
@@ -29,7 +28,10 @@ namespace AppleKraken
...
+                ->Field("MaxCameraDistance", &FollowingCameraComponent::m_max_camera_distance)
```
Comment (michalpelka) [inline Project/Gem/Source/KrakenCamera/FollowingCameraComponent.cpp:33]: Convention - please use camel-case instead '_'

File: Project/Gem/Source/KrakenCamera/FollowingCameraComponent.cpp [hunk 4]
```diff
@@ -29,7 +28,10 @@ namespace AppleKraken
...
+                ->Field("LookAtCenterEntity", &FollowingCameraComponent::m_look_at_center_entity);
```
Comment (michalpelka) [inline Project/Gem/Source/KrakenCamera/FollowingCameraComponent.cpp:34]: Convention - please use camel-case instead '_'

File: Project/Gem/Source/KrakenCamera/FollowingCameraComponent.cpp [hunk 5]
```diff
@@ -72,57 +95,82 @@ namespace AppleKraken
...
             AZ_Warning("FollowingCameraComponent", false, "m_target is empty!");
         }
         if (!m_isActive)
         {
             return;
         }
 
-        AZ::Transform target_world_transform;
-        EBUS_EVENT_ID_RESULT(target_world_transform, m_target, AZ::TransformBus, GetWorldTM);
+        if (!m_activated)
```
Comment (michalpelka) [inline Project/Gem/Source/KrakenCamera/FollowingCameraComponent.cpp:107]: I do not understand why execute this code only if it is deactivated.

File: Project/Gem/Source/KrakenCamera/FollowingCameraComponent.cpp [hunk 6]
```diff
@@ -149,11 +197,16 @@ namespace AppleKraken
...
+        {
+            m_max_camera_distance = m_min_camera_distance;
+        }
+
         const AzFramework::InputChannelId& channelId = inputChannel.GetInputChannelId();
         // TODO take into account the refresh rate on key pressed for smooth experience
         if (channelId == AzFramework::InputDeviceKeyboard::Key::AlphanumericW)
         {
-            m_zoomChange = AZStd::min(m_zoomMax, m_zoomChange + m_zoomSpeed);
+            m_zoomChange = AZStd::min((-m_min_camera_distance + 2.0f), m_zoomChange + m_zoomSpeed);
```
Comment (michalpelka) [inline Project/Gem/Source/KrakenCamera/FollowingCameraComponent.cpp:209]: It would be better to set limits in reflection than adding here 2.0f.

Comment (pawelbudziszewski) [reply Project/Gem/Source/KrakenCamera/FollowingCameraComponent.cpp]: This was done to tune the `zero` value of the min distance. It's more intuitive if the closest possible distance set in the UI is 0.0, not -2.0

**Comment by pawelbudziszewski** [comment]
I found some problems with the camera component, which makes the code in this PR work incorrectly sometimes.

**Comment by pawelbudziszewski** [comment]
There was a bug in the camera's initial transform recording. Because of it, the camera was translated and rotated according to the target transform. Now it should work fine.
